calico_containers/tests/st/docker_host.py

```diff
@@ -15,16 +15,21 @@ def __init__(self, name):
         pwd = sh.pwd().stdout.rstrip()
         docker.run("--privileged", "-v", pwd+":/code", "--name", self.name, "-tid", "jpetazzo/dind")
```

Comment:
I don't think we're using Docker very well when we use this design pattern.  We go to all the trouble of using Docker to be able to create containerized builds and ST runs, and then we just mount our whole repo into the container.

Instead, what we should be doing is using jpetazzo/dind as a base image, then build a new Docker image that has the code to be tested copied in and the inner Docker images preloaded.

The only volumes we should need to mount are either 

a) persistent storage (don't see why we'd need that for ST)
b) output directories for logs, coverage, etc.

This comment also applies to the way we do builds and UTs, but I'm happy to raise a separate issue for those.